### PR TITLE
Moving esConsumes statement from the analyze method into the constructor

### PR DESCRIPTION
**Pull request by igv4321**
#### PR description:

Moving esConsumes statement from the "analyze" method into the constructor.

#### PR validation:

Checked that the database read/write sequence for HFPhase1PMTParams objects works as described in CondTools/Hcal/test/readme_HFPhase1PMTParams.txt

This code is not used by any workflows (calibration tables preparation only), so default matrix tests were not run.

### CODE REVIEW

File: CondTools/Hcal/interface/BoostIODBReader.h [hunk 1]
```diff
@@ -51,16 +51,16 @@ class BoostIODBReader : public edm::one::EDAnalyzer<> {
   void analyze(const edm::Event&, const edm::EventSetup&) override;
 
   std::string outputFile_;
+  edm::ESGetToken<DataType, RecordType> tok_;
```
Comment (tvami) [inline CondTools/Hcal/interface/BoostIODBReader.h:54]:
```suggestion
  const edm::ESGetToken<DataType, RecordType> tok_;
```

**Comment by tvami** [comment]
Hi @igv4321 thanks for doing this change! 
I believe `cmsRun HFPhase1PMTParamsDBWriter_cfg.py` and the corresponding readers could easily be made into a unit test, and I think we should do it. Have you done a unit test for CMSSW before? Should I point you to some example? Thanks.

File: CondTools/Hcal/interface/BoostIODBReader.h (hunk 1 above)
Comment (igv4321) [reply CondTools/Hcal/interface/BoostIODBReader.h]: OK, changed as suggested.

**Comment by igv4321** [comment]
@tvami I don't remember writing unit tests... Yes, please point out some example.

**Comment by tvami** [comment]
@igv4321 
here is an example, the `BuildFile`
https://github.com/cms-sw/cmssw/blob/master/CondTools/SiPixel/test/BuildFile.xml
launches the test `createTestDBObjects.sh`
which then does the `cmsRun`
https://github.com/cms-sw/cmssw/blob/master/CondTools/SiPixel/test/createTestDBObjects.sh#L7-L8

And then a next line could read the sqlite file being created by the `HFPhase1PMTParamsDBWriter_cfg`

**Comment by igv4321** [comment]
@tvami Ok, thanks. Unit test added.

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
+db
 * new unit test passes: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d93e44/24117/unitTests/src/CondTools/Hcal/test/HFPhase1PMTParams_unittest/testing.log

**Comment by perrotta** [comment]
+1

**Comment by qliphy** [comment]
@igv4321 there is a unit test failure in IB after this PR gets merged.  Please have a check and fix it:

https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_amd64_gcc10/CMSSW_12_4_X_2022-04-22-2300/unitTestLogs/CondTools/Hcal#/19-19
ata/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_12_4_X_2022-04-22-2300/src/CondTools/Hcal/test/testHFPhase1PMTParams.sh: line 6: /data/cmsbld/jenkins/workspace/ib-run-qa/CMSSW_12_4_X_2022-04-22-2300/test/slc7_amd64_gcc10/write_HFPhase1PMTParams: No such file or directory
Failure running write_HFPhase1PMTParams: status 127
status = 32512

---> test HFPhase1PMTParams_unittest had ERRORS
TestTime:1
^^^^ End Test HFPhase1PMTParams_unittest ^^^^

**Comment by igv4321** [comment]
I can't reproduce it, but looking at the log file, this problem could be due to an out-of-order execution of tests in the production environment. @tvami  Is there some way to ensure that the "<bin file="write_HFPhase1PMTParams.cc">" branch of CondTools/Hcal/test/BuildFile.xml (which actually builds the "write_HFPhase1PMTParams" executable) is completed before the "<bin file="HFPhase1PMTParams_unittest.cc">" branch is run?

**Comment by mmusich** [comment]
> Is there some way to ensure that the <bin file="write_HFPhase1PMTParams.cc"> branch of CondTools/Hcal/test/BuildFile.xml (which actually builds the "write_HFPhase1PMTParams" executable) is completed before the <bin file="HFPhase1PMTParams_unittest.cc"> branch is run?
 
The flag `PRE_TEST` should do the trick:

e.g.:
https://github.com/cms-sw/cmssw/blob/d30234439f449909c211a85d5e5c8b5554a0f7f7/Alignment/OfflineValidation/test/BuildFile.xml#L8

**Comment by igv4321** [comment]
OK, thanks. The fix is provided in the PR #37666

**Comment by tvami** [comment]
Thanks @igv4321 for the fix.
@qliphy why didnt the unit test in here https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d93e44/24117/unitTests/src/CondTools/Hcal/test/HFPhase1PMTParams_unittest/testing.log catch this? Should I test the bugfix PR differently? (for example with the full CMSSW setup?)

**Comment by smuzaffar** [comment]
@tvami , for PR tests and local tests `${CMSSW_BASE}/test/${SCRAM_ARCH}/write_HFPhase1PMTParams` exists but for IB tests it does not. For IB tests, we just create a cmssw dev area and all the tests executables are in $CMSSW_RELEASE_BASE.   As for unit tests the test/arch directories are in PATH so just run `write_HFPhase1PMTParams`

**Pull request merged**